### PR TITLE
[FLINK-6817] [table] Fix NPE when preceding is not set in OVER window

### PR DESCRIPTION
**Pull request by wuchong**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/windows.scala [hunk 1]
```diff
@@ -91,6 +91,10 @@ class OverWindowWithOrderBy(
     */
   def as(alias: Expression): OverWindow = {
 
+    if (null == preceding) {
```
Comment (sunjincheng121) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/windows.scala]: For the strict system type, I recommend adding the `OverWindowWithPreceding` class to guide the user to do the correct operation.

Comment (wuchong) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/windows.scala]: Thank you. I think it's a good idea.

**Comment by wuchong** [comment]
@sunjincheng121  I've added a `OverWindowWithPreceding` class

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/windows.scala [hunk 2]
```diff
@@ -98,7 +98,7 @@ object Over {
     */
   def orderBy(orderBy: String): OverWindowWithOrderBy = {
     val orderByExpr = ExpressionParser.parseExpression(orderBy)
-    new OverWindowWithOrderBy(Seq[Expression](), orderByExpr)
+    new OverWindowWithOrderBy(Array[Expression](), orderByExpr)
```
Comment (sunjincheng121) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/windows.scala]: Good catch!
And suggest to change the `java/windows.scala` as well. What do you think?

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/windows.scala [hunk 3]
```diff
@@ -127,3 +127,21 @@ class PartitionedOver(private val partitionByExpr: Array[Expression]) {
     new OverWindowWithOrderBy(partitionByExpr, orderByExpr)
   }
 }
+
+
+class OverWindowWithOrderBy(
+  private val partitionByExpr: Array[Expression],
```
Comment (sunjincheng121) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/windows.scala]: Suggest that change `partitionByExpr` to `partitionBy` for keep param name consistent with SCALA. What do you think?

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/windows.scala [hunk 4]
```diff
@@ -127,3 +127,21 @@ class PartitionedOver(private val partitionByExpr: Array[Expression]) {
     new OverWindowWithOrderBy(partitionByExpr, orderByExpr)
```
Comment (sunjincheng121) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/windows.scala]: Suggest that change `partitionByExpr` to `partitionBy` for keep param name consistent with SCALA. What do you think?

**Comment by twalthr** [comment]
I'm fine with this change. I think we should also include this in 1.3.1, what do you think @fhueske?

**Comment by fhueske** [comment]
Yes, looks good to me as well. +1 for merging and including in 1.3.1

**Comment by wuchong** [comment]
merging...

**Pull request merged**